packages/tailwindcss/src/variants.ts

```diff
@@ -899,8 +899,8 @@ export function createVariants(theme: Theme): Variants {
   staticVariant('portrait', ['@media (orientation: portrait)'], { compounds: false })
   staticVariant('landscape', ['@media (orientation: landscape)'], { compounds: false })
 
-  staticVariant('ltr', ['&:where([dir="ltr"], [dir="ltr"] *)'])
-  staticVariant('rtl', ['&:where([dir="rtl"], [dir="rtl"] *)'])
+  staticVariant('ltr', ['&:where(:dir(ltr), :dir(ltr) *, [dir="ltr"], [dir="ltr"] *)'])
+  staticVariant('rtl', ['&:where(:dir(rtl), :dir(rtl) *, [dir="rtl"], [dir="rtl"] *)'])
```

Comment:
Don't need `:dir(rtl) *` here actually because the pseudo-class inherits already in the browser 🙌🏻

```suggestion
  staticVariant('ltr', ['&:where(:dir(ltr), [dir="ltr"], [dir="ltr"] *)'])
  staticVariant('rtl', ['&:where(:dir(rtl), [dir="rtl"], [dir="rtl"] *)'])
```